examples/DistributedTraining.cpp

```diff
@@ -30,7 +30,8 @@ int main() {
   auto reducer = std::make_shared<fl::CoalescingReducer>(
       /*scale=*/1.0 / worldSize,
       /*async=*/true,
-      /*contiguous=*/true);
+      /*contiguous=*/true,
+      /*useMixedPrecision*/ false);
```

Comment:
Instead of passing this bool flag or even an `af::dtype`, let's do what I suggested above for autograd and Modules - let's look at the `af::dtype` if the `af::array` which we're synchronizing and choose which NCCL param type we use based on that so there aren't any unexpected casts.